opm/models/blackoil/blackoilconvectivemixingmodule.hh

```diff
@@ -382,6 +375,53 @@ public:
     }
 };
 
+/*!
+ * \ingroup BlackOil
+ * \class Opm::BlackOilConvectiveMixingIntensiveQuantities
+ *
+ * \brief Provides the volumetric quantities required for the equations needed by the
+ *        convective mixing (DRSDTCON) model.
+ */
+template <class TypeTag, bool enableConvectiveMixingV = getPropValue<TypeTag, Properties::EnableConvectiveMixing>()>
+class BlackOilConvectiveMixingIntensiveQuantities
+{
+    using Implementation = GetPropType<TypeTag, Properties::IntensiveQuantities>;
+    using Evaluation = GetPropType<TypeTag, Properties::Evaluation>;
+    using FluidSystem = GetPropType<TypeTag, Properties::FluidSystem>;
+
+public:
+    /*!
+     * \brief Compute the intensive quantities needed to handle convective dissolution
+     *
+     */
+    void updateSaturatedDissolutionFactor_()
+    {
+        auto& fs = asImp_().fluidState();
```

Comment:
Another tiny nit: The `fs` object is used in just a single location&ndash;the call to `saturatedDissolutionFactor()`&ndash;so I guess you could just as well use `asImp_().fluidState()` directly in that function call.